code/obj/machinery/computer/cloning.dm

```diff
@@ -122,23 +123,36 @@
 
 	else if (isscrewingtool(W))
 		playsound(src.loc, "sound/items/Screwdriver.ogg", 50, 1)
-		if(do_after(user, 2 SECONDS))
-			if(src.status & BROKEN)
-				boutput(user, "<span class='notice'>The broken glass falls out.</span>")
+		if (do_after(user, 2 SECONDS))
+			logTheThing("station", user, null, "disconnects the cloning console at [log_loc(src)].")
```

Comment:
If logging is the only thing different (possible there's other differences, this seems to be it at a cursory skim), maybe that's an indication that there should be a utility proc for deconstructing instead that would allow a specific type to call it to determine if the thing was deconstructed and (if so) do other stuff (logging, in this case)?

Reply:
There are a few other changes for cloning. It saves the records to the circuitboard and drops the various upgrades. I'll try out your approach.

Reply:
Could have a stub proc on the parent for additional deconstruct stuff, then call that as part of the deconstruct workflow on the generic parent. This would let computers that do more than just turn into a frame + circuitboard inject their additional behavior by overriding the proc.

Reply:
Done